version: '3.7'
services:
  app:
    image: docker.pkg.github.com/yanislav-igonin/bubble-face-bot/bubble-face-bot:latest
    networks:
      - web
    environment: 
      - NODE_ENV=production
      - BOT_TOKEN=${BOT_TOKEN}
      - WEBHOOK_HOST=${WEBHOOK_HOST}
      - WEBHOOK_PORT=${WEBHOOK_PORT}
      - WEBHOOK_PATH=${WEBHOOK_PATH}
      - IS_WEBHOOK_DISABLED=false
      - IMAGINARY_HOST=${IMAGINARY_HOST}
      - IMAGINARY_API_KEY=${IMAGINARY_API_KEY}
    deploy:
      replicas: 1
      update_config:
        parallelism: 1
        delay: 3s
        failure_action: rollback
        order: stop-first
      restart_policy:
        condition: on-failure
        window: 10s
      placement:
        constraints:
          - node.role == manager
      labels:
        traefik.enable: "true"
        traefik.docker.network: "web"

        traefik.http.services.bubble-face-bot.loadbalancer.server.port: ${WEBHOOK_PORT}

        traefik.http.routers.bubble-face-bot-secure.entrypoints: "https"
        traefik.http.routers.bubble-face-bot-secure.rule: "Host(`${WEBHOOK_URL}`) && PathPrefix(`${WEBHOOK_PATH}`)"
        traefik.http.routers.bubble-face-bot-secure.tls: "true"
        traefik.http.routers.bubble-face-bot-secure.service: "bubble-face-bot"

networks:
  web:
    driver: overlay
    external: true
